src/linux/Target.zig

```diff
@@ -0,0 +1,80 @@
+const std = @import("std");
+
+pub const Target = @This();
+
+allocator: std.mem.Allocator,
+pid: std.os.pid_t,
+
+pub fn init(allocator: std.mem.Allocator) !Target {
+    return Target{
+        .allocator = allocator,
+        .pid = undefined,
+    };
+}
+
+pub fn deinit(self: *Target) void {
+    _ = self;
+}
+
+pub fn attach(self: *Target, pid: usize) !void {
+    _ = self;
+    _ = pid;
+}
+
+pub fn spawn(self: *Target, args: []const []const u8) !void {
+    const pid = try std.os.fork();
+    if (pid == 0) { // child
+        _ = try ptrace(PTRACE_TRACEME, 0, null, null);
+
+        const argv_buf = try self.allocator.allocSentinel(?[*:0]u8, args.len, null);
+        for (args, 0..) |arg, i| argv_buf[i] = (try self.allocator.dupeZ(u8, arg)).ptr;
+
+        _ = std.os.execvpeZ_expandArg0(.expand, argv_buf.ptr[0].?, argv_buf.ptr, @ptrCast([*:null]?[*:0]u8, std.os.environ.ptr)) catch {};
+    } else {
+        self.pid = pid;
+        std.debug.print("{}", .{std.os.waitpid(pid, 0)});
+    }
+}
+
+pub fn @"suspend"(self: *Target) !void {
+    _ = self;
+}
+
+pub fn @"resume"(self: *Target) !void {
+    _ = try ptrace(PTRACE_CONT, self.pid, null, null);
+}
+
+pub fn kill(self: *Target) !void {
+    _ = self;
+}
+
+const PTRACE_TRACEME = 0;
```

Comment:
Please note that Andrew recently merged a PR with `ptrace` in it for Linux. I will also add a common wrapper for Linux and darwin shortly available in `os.zig`.